include/deal.II/lac/trilinos_sparse_matrix.h

```diff
@@ -2223,10 +2225,33 @@ namespace TrilinosWrappers
         * inverse operations.
         * The <tt>vmult</tt> and <tt>Tvmult</tt> operations retain the standard
         * definitions inherited from @p op.
+        *
+        * @note This function is enabled only if the solver and preconditioner
+        * derive from the respective TrilinosWrappers base classes.
         */
         template <typename Solver, typename Preconditioner>
-        TrilinosPayload inverse_payload (Solver &, const Preconditioner &) const;
+        typename std::enable_if<
+        std::is_base_of<TrilinosWrappers::SolverBase,Solver>::value and
+        std::is_base_of<TrilinosWrappers::PreconditionBase,Preconditioner>::value,
+            TrilinosPayload>::type
+            inverse_payload (Solver &, const Preconditioner &) const;
 
+        /**
+        * Returns a payload configured for inverse operations
+        *
+        * Invoking this constructor will configure two additional functions,
+        * namely <tt>inv_vmult</tt> and <tt>inv_Tvmult</tt>, both of which
+        * are disabled because the @p Solver or @p Preconditioner are not
+        * compatible with Epetra_MultiVector.
+        * The <tt>vmult</tt> and <tt>Tvmult</tt> operations retain the standard
+        * definitions inherited from @p op.
+        */
+        template <typename Solver, typename Preconditioner>
+        typename std::enable_if<
+        not (std::is_base_of<TrilinosWrappers::SolverBase,Solver>::value and
+             std::is_base_of<TrilinosWrappers::PreconditionBase,Preconditioner>::value),
+                 TrilinosPayload>::type
```

Comment:
In some sense, what you have here is a set of "concepts" the template arguments have to satisfy. Can you reflect this in the documentation, e.g., by saying "The C++ compiler will only consider this function if (i)..., (ii)..., and (iii) ... are satisfied"